custom/icds/location_rationalization/parser.py

```diff
@@ -0,0 +1,74 @@
+from collections import defaultdict
+
+from custom.icds.location_rationalization.const import (
+    EXTRACT_TRANSITION,
+    MERGE_TRANSITION,
+    MOVE_TRANSITION,
+    SPLIT_TRANSITION,
+    TRANSITION_COLUMN_NAME,
+    VALID_TRANSITIONS,
+)
+
+
+class Parser(object):
+    def __init__(self, worksheet, location_types):
+        self.worksheet = worksheet
+        self.requested_transitions = {}
+        self.errors = []
+        self.location_types = location_types
+        self.transitions = {location_type: {
+            MERGE_TRANSITION: defaultdict(list),
+            SPLIT_TRANSITION: defaultdict(list),
+            MOVE_TRANSITION: {},
+            EXTRACT_TRANSITION: {},
+        } for location_type in location_types}
+
+    def parse(self):
+        for row in self.worksheet:
+            transition = row.get(TRANSITION_COLUMN_NAME)
+            if not transition:
+                continue
+            if transition not in VALID_TRANSITIONS:
+                self.errors.append("Invalid Transition %s" % transition)
+            self._parse_row(row)
+        return self.transitions, self.errors
+
+    def _parse_row(self, row):
+        transition = row.get(TRANSITION_COLUMN_NAME)
+        for location_type in self.location_types:
```

Comment:
Should this function validate that the given locations are active?

Reply:
Hmm, yes that would be a valuable addition
https://github.com/dimagi/commcare-hq/pull/26865/commits/3260ed6b08e4ef11f60a8790654e54ecdf81d429